src/riscv-integration.adoc

```diff
@@ -774,15 +774,17 @@ Environment call +
 Environment break +
 Load/store/AMO address breakpoint
 
-| .>| *{cheri_excep_mcause}* .<| *CHERI faults due to:* +
+| .>| *{cheri_excep_mcause}* +
+*{cheri_excep_mcause}* .<| *CHERI faults due to:* +
 *PCC <<asr_perm>> clear* +
 *Branch/jump target address checks (tag, execute permissions and bounds)*
 
-| .>|*{cheri_excep_mcause}* .<|*Prior to address translation for an explicit memory access:* +
-*Load/store/AMO capability address misaligned* +
-*CHERI fault due to capability checks (tag, permissions and bounds)*
 | .>|4,6 .<|Optionally: +
 Load/store/AMO address misaligned
```

Comment:
The motivation for putting misaligned faults after CHERI is to avoid splitting standard RISC-V exception causes in 2 blocks i.e. we now have
* RISC-V exception (misaligned)
* CHERI exceptions
* Other RISC-V exceptions

This is not ideal for some hardware implementations where all the RISC-V load/store/amo exceptions (including misaligned) come from one hardware block while the CHERI exceptions come from another block. In the original case, this is not a problem because the two hardware blocks produce RISC-V and CHERI exceptions separately and then we have a 2-way priority (if !CHERI_exception then RISC-V_exception else CHERI_exception).

My preference is to revert this change.

Reply:
Alignment faults must be lower priority than CHERI faults. Otherwise you have to reimplement the entirety of the CHERI checks in your misaligned access emulation routine. Which people will 100% screw up.

Reply:
Thanks for the comment @jrtc27! I was unaware of those existing checks. Seems that's an even stronger reason than what I pointed out ifrom a hardware implementation perspective!